### PR TITLE
Made result::error movable.

### LINKED ISSUE #108: make result::error accessible by reference or move
When using result with custom types, the error might be non-trival, but hold on to resources, such a strings. In those cases it would be very useful if the error could be obtained without a copy.

### PR DESCRIPTION
**Pull request by klemens-morgenstern**
Closes #108.

### CODE REVIEW

**Comment by pdimov** [comment]
I don't see why you seem to have made `XM` copyable. The test is supposed to test a move-only type.

File: test/result_error_move.cpp [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+using namespace boost::system;
+
+struct XM
+{
+    int v_;
+
+    explicit XM( int v = 0 ): v_( v ) {}
+
+    XM( XM const& ) = default;
```
Comment (klemens-morgenstern) [inline test/result_error_move.cpp:18]:
```suggestion
    XM( XM const& ) = delete;
```

This should've been `delete`.

**Comment by pdimov** [comment]
You can't move from const objects. These tests only worked because of the copy constructor.

**Comment by klemens-morgenstern** [comment]
yeah, i missed the right-`const`.

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -163,3 +163,4 @@ boost_test(TYPE run SOURCES result_emplace.cpp)
 boost_test(TYPE run SOURCES result_error_construct4.cpp)
 boost_test(TYPE run SOURCES result_value_construct4.cpp)
 boost_test(TYPE run SOURCES result_value_construct5.cpp)
+boost_test(TYPE run SOURCES result_error_move.cpp)
```
Comment (pdimov) [inline test/CMakeLists.txt:166]: A newline is missing here at the end of file.

**Pull request merged**